@cnauroth `sun.misc.Signal` class exists in both JDK 8 and JDK 9+, but it is not a public API, see https://bugs.openjdk.org/browse/JDK-8149750.

If we don't pursue compiling Hadoop using a modern JDK with `-release 8`(see below), we don't need this patch.

```
export JAVA_HOME=/path/of/openjdk-17
mvn compile -DskipTests -Djavac.version=8 -Dmaven.compiler.release=8
```

```
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project hadoop-common: Compilation failure: Compilation failure: 
[ERROR] /Users/chengpan/Projects/apache-hadoop/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/SignalLogger.java:[22,16] package sun.misc does not exist
[ERROR] /Users/chengpan/Projects/apache-hadoop/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/SignalLogger.java:[23,16] package sun.misc does not exist
[ERROR] /Users/chengpan/Projects/apache-hadoop/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/SignalLogger.java:[44,43] cannot find symbol
[ERROR]   symbol:   class SignalHandler
[ERROR]   location: class org.apache.hadoop.util.SignalLogger
...
```

--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/service/launcher/IrqHandler.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/service/launcher/IrqHandler.java
@@ -37,7 +37,6 @@ import org.apache.hadoop.classification.InterfaceStability;
  */
 @InterfaceAudience.Private
 @InterfaceStability.Unstable
-@SuppressWarnings("UseOfSunClasses")
 public final class IrqHandler implements SignalUtil.Handler {
   private static final Logger LOG = LoggerFactory.getLogger(IrqHandler.class);
   

--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/SignalUtil.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/SignalUtil.java
@@ -63,7 +63,7 @@ public class SignalUtil {
     private static final DynMethods.UnboundMethod GET_NAME_UNBOUND_METHOD =
         new DynMethods.Builder("getName").impl(JDK_SIGNAL_CLAZZ).build();
 
-    private final Object/* sun.misc.SignalHandler */ delegate;
+    private final Object/* sun.misc.Signal */ delegate;
     private final DynMethods.BoundMethod getNumberMethod;
     private final DynMethods.BoundMethod getNameMethod;
 
@@ -91,6 +91,7 @@ public class SignalUtil {
       return getNameMethod.invoke();
     }
 
+    @Override
     public boolean equals(Object obj) {
       if (this == obj) {
         return true;
@@ -101,10 +102,12 @@ public class SignalUtil {
       return false;
     }
 
+    @Override
     public int hashCode() {
       return delegate.hashCode();
     }
 
+    @Override
     public String toString() {
       return delegate.toString();
     }
@@ -117,7 +120,7 @@ public class SignalUtil {
 
   static class JdkSignalHandlerImpl implements Handler {
 
-    private final Object/* sun.misc.Signal */ delegate;
+    private final Object/* sun.misc.SignalHandler */ delegate;
     private final DynMethods.BoundMethod jdkSignalHandlerHandleMethod;
 
     JdkSignalHandlerImpl(Handler handler) {

